### ISSUE TITLE
Bezier curve issue with line alignment in Canvas

### ISSUE THREAD

**Issue by ymegane** [pixijs/pixijs#8056]
### Expected Behavior
The result of drawing a shape with bezier curve in WebGL and Canvas should be the same.

### Current Behavior
If line alignment is not 0.5, there will be a difference in the drawing result.(left: WebGL, right: Canvas)

alignment: 1
![image](https://user-images.githubusercontent.com/2368078/146526814-1d288836-afa9-436a-a1ae-614426120dad.png)

alignment: 0.5
![image](https://user-images.githubusercontent.com/2368078/146528476-ff83bea5-648e-474f-b70d-9f846ab03b12.png)

### Possible Solution
I don't fully understand the recurrence condition, but it looks like the problem occurs when the shape to be drawn contains the same coordinates.

After a bit of debugging, it seems that division by zero can occur here, and this may be the cause.

https://github.com/pixijs/pixijs/blob/f9d526704cd6e992e9194ff745be906a556c66cb/packages/canvas/canvas-graphics/src/utils/PolygonUtils.ts#L39-L42
https://github.com/pixijs/pixijs/blob/f9d526704cd6e992e9194ff745be906a556c66cb/packages/canvas/canvas-graphics/src/utils/PolygonUtils.ts#L55-L58

### Steps to Reproduce
https://codesandbox.io/s/pixi-canvas-beziercurve-problem-xlvfc?file=/src/shapeHelper.ts

### Environment

- **`pixi.js` version**: 6.2.0
- **Browser & Version**: 96.0.4664.110（Official Build） （x86_64）
- **OS & Version**: macOS 12.1
- **Running Example**: https://codesandbox.io/s/pixi-canvas-beziercurve-problem-xlvfc

**Comment by bigtimebuddy** [comment]
I agree this seems inconsistent, but in general it is better practice to not repeat the starting point when drawing a closed shape. Instead use closePath() to finish the shape if possible. Over-lapping points can create funny issues for strokes too.

I'm guessing webgl just handles this situation more gracefully. Fix welcome for this.

**Comment by ymegane** [comment]
The problem in my project was solved by using closePath(). Thank you for the instruction!

It would certainly be better if it worked the same way as WebGL, but it doesn't seem to be a major problem.
I'll try to fix that some other time!
